### DISCUSSION TITLE
Updating `tickLength` based on state

### DISCUSSION THREAD

**Discussion by jbrundagejr** [airbnb/visx#1827] [General]
Hey all- I don't know if I just haven't found the right prop, but I'm trying to get my `tickLength` to update based on a state variable. Is there a way of accessing the tick value? Something like `<AxisBottom tickLength={tickX === tickValue ? 20 : 10}/>` where tickX is my state variable being updated. Thanks in advance for any help!!

**Comment by williaster** [comment]
Hey @jbrundagejr - currently `tickLength` is a property shared across all ticks and can only be a [`number` type](https://github.com/airbnb/visx/blob/master/packages/visx-axis/src/types.ts#L93). we would need to update this to be like [`tickLabelProps`](https://github.com/airbnb/visx/blob/master/packages/visx-axis/src/types.ts#L24), which takes an object or a function, in order to support per-tick customization.
